BDArmory/Control/BDModulePilotAI.cs

```diff
@@ -2329,6 +2333,10 @@ void FlyToTargetVessel(FlightCtrlState s, Vessel v)
                         magnifier += ((magnifier - 1f) * Mathf.Sin(Time.time * 0.75f));
                         if (BDArmorySettings.DEBUG_TELEMETRY || BDArmorySettings.DEBUG_AI) debugString.AppendLine($"targetAngVel: {targetAngVel:F4}, magnifier: {magnifier:F2}");
                         target -= magnifier * leadOffset; // The effect of this is to exagerate the lead if the angular velocity is > 1
+                        //need to subtract the angle of the gun if offset/angled from the target est. position
+                        //leadOffset = (vesselTransform.position + (vesselTransform.up * distanceToTarget)) - (weapon.fireTransforms[0].transform.position + (weapon.fireTransforms[0].forward * distanceToTarget));
+                        leadOffset = (weapon.fireTransforms[0].transform.position + (weapon.fireTransforms[0].forward * distanceToTarget)) - (vesselTransform.position + (vesselTransform.up * distanceToTarget));
```

Comment:
hmm. In testing it did manage to line up shots with guns pointing 90deg sideways broadside or 90deg straight up (not well, mind you, the Aeris3A doesn't have the greatest Yaw + Roll auth, and the moment the target started actively maneuvering...).
Will try the Quaternion method.

Reply:
Here's what I mean is happening if you use vector addition instead of rotations:
![leadoffset](https://github.com/BrettRyland/BDArmory/assets/18440126/08bd90d4-f239-446a-bff0-54da48800bc7)
For small angles you get the angles A and B about the same (but not exactly), so the yellow gun would point close to the right direction if the plane points along the green line, whereas for larger angles you get that C and D are very different so the red gun wouldn't be pointing in the right direction if the plane points along the pink line.

Reply:
Hmm. testing with the quaternion method I must be doing something wrong; testing just has the target be directly ahead of the craft as if it were the old vesselTransform.up aiming, Also not sure if that diagram matches experimental results; AI was happily piloting a craft with a broadside vulcan mounted at a 75deg angle off prograde (so close to 'C') and correctly leading target under the vector method?

Reply:
Hmm. Something odd is going on here. Not sure if I've implemented the quaternions wrong, the test craft is still flying as if the weapons were mounted straight forward and calculating lead from vesselTransform.up. Adding to the confusion is testing yesterday had craft with weapons mounted a75deg from prograde broadside orientation similar to "C" in the diagram properly lining up the target under the vector method?